src/main/java/pl/put/poznan/transformer/rest/JSONToolsController.java

```diff
@@ -5,96 +5,132 @@
-            }
-            if(jsonNode.size() != 1){
-                return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
-            }
-            try{
-                return ResponseEntity.status(200).body(service.minify(json));
-            }catch(Exception exception){
-                logger.error(exception.toString());
-                return ResponseEntity.status(500).body("{\"error\":\"Internal error!\"}");
-            }
+        }
+
+        if(jsonNode.size() != 1){
+            return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
+        }
+
+        JsonNode json;
+        try{
+            json = getInput(jsonNode);
+        }catch(JSONException exception){
+            return ResponseEntity.status(400).body("{\"error\":\"Missing or broken 'json' property!\"}");
+        }
+
+        try{
+            return ResponseEntity.status(200).body(tool.decorate(new JSONObject(json.toString())).getJson());
+        }catch(Exception exception){
+            logger.error(exception.toString());
+            return ResponseEntity.status(500).body("{\"error\":\"Internal error!\"}");
         }
     }
 
     @RequestMapping(value="pretty", method = { RequestMethod.POST }, produces = "application/json")
     public ResponseEntity<String> pretty(
             @RequestBody String payload
     ) {
+        final JSONToolPrettify tool = new JSONToolPrettify();
+
         logger.info("POST /api/v1/pretty");
-        JsonNode jsonNode = service.parse(payload);
-        if(jsonNode == null){
+        JsonNode jsonNode;
+
+        try{
+            jsonNode = parse(payload);
+        }catch(JSONException exception){
             return ResponseEntity.status(400).body("{\"error\":\"Payload is not valid JSON!\"}");
-        }else{
-            JsonNode json = service.getInput(jsonNode);
-            if(json == null){
-                return ResponseEntity.status(400).body("{\"error\":\"Missing or broken 'json' property!\"}");
-            }
-            if(jsonNode.size() != 1){
-                return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
-            }
-            try{
-                return ResponseEntity.status(200).body(service.pretty(json));
-            }catch(Exception exception){
-                logger.error(exception.toString());
-                return ResponseEntity.status(500).body("{\"error\":\"Internal error!\"}");
-            }
         }
+
+        if(jsonNode.size() != 1){
+            return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
+        }
+
+        JsonNode json;
+        try{
+            json = getInput(jsonNode);
+        }catch(JSONException exception){
+            return ResponseEntity.status(400).body("{\"error\":\"Missing or broken 'json' property!\"}");
+        }
+
```

Comment:
Dużo tu jest copy paste może da się to zamknąć w jakiejś klasie np ResponseEntityJSONTool i daćdo niej 
return ResponseEntity.status(400).body("{\"error\":\"Payload is not valid JSON!\"}");
return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
return ResponseEntity.status(400).body("{\"error\":\"Missing or broken 'json' property!\"}");

i jescze drugą ResponseEntityJSONToolsFilter która by jeszcze miała to
return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' and 'keys' properties!\"}");